### ISSUE TITLE
`vue/valid-next-tick` is reported when use nested callback + `nextTick`

### ISSUE THREAD

**Issue by y-yagi** [vuejs/eslint-plugin-vue#1936]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version: 8.20.0**
- **eslint-plugin-vue version: 9.2.0** 
- **Node version: v16.13.2** 

**Please show your full configuration:**

```json5
{
  "extends": [
    "eslint:recommended",
    "eslint",
    "plugin:vue/recommended"
  ],
  "parserOptions": {
    "parser": "@typescript-eslint/parser",
    "sourceType": "module",
    "project": "./tsconfig.json",
    "extraFileExtensions": [".vue", ".snap"]
  },
  "plugins": [
    "@typescript-eslint"
  ],
  "env": {
    "browser": true,
    "node": true,
    "jest": true,
    "jquery": true,
    "es6": true
  },
}
```

**What did you do?**	

```vue
this.$watch('time', () => this.$nextTick(() => update()));
```

**What did you expect to happen?**

No errors

**What actually happened?**

"Either await the Promise or pass a callback function to `nextTick`  vue/valid-next-tick" is reported.

**Repository to reproduce this issue**

**Comment by y-yagi** [comment]
It worked well. Thank you!!
